### PR TITLE
Formatting lints for `=@` and `else if`s

### LINKED ISSUE #649: lint expressions of the form `a =- b` (note the formatting!)...
...which is likely a typo and should be either `a -= b` or `a = -b`. Similar for `a =! b`.

Using the snippet method on an assignment which assigns ExprUnary and checking if it contains either `=-` or `=!` should be easy enough.

### LINKED ISSUE #718: lint deceiving formatting: if { .. } if { .. }
If there are two consecutive if-expressions in a block and the second begins on the same line the first ends, the programmer may have wanted to do an `else if`, but forgot the `else`. Suggest to add either the `else` or a line break between the statements to reduce confusion. (Nicked from GCC6)

Example:
```
fn main() {
	let x = 0;
	if x == 0 {
		println!("Hello");
	} if x == 1 { //~ERROR please insert a line break or an `else`
		println!("World!");
	}
}
```

### PR DESCRIPTION
**Pull request by mcarton**
Fix #649. Fix #718.

### CODE REVIEW

File: src/consts.rs [hunk 1]
```diff
@@ -206,7 +206,6 @@ fn lit_to_constant(lit: &LitKind) -> Constant {
 }
 
 fn constant_not(o: Constant) -> Option<Constant> {
-    use syntax::ast::LitIntType::*;
```
Comment (llogiq) [inline src/consts.rs]: Good cleanup. :+1:

**Comment by llogiq** [comment]
That was fast. :smile: @mcarton I'd like to have you writing some more challenging lints and leave the easy ones to new contributors. OK?

**Comment by mcarton** [comment]
There are plenty of [easy lints left](https://github.com/Manishearth/rust-clippy/issues?q=is:issue+is:open+label:E-easy), I won’t implement them all, promised :stuck_out_tongue_winking_eye:.

**Comment by llogiq** [comment]
:smile: OK then.

**Comment by Manishearth** [comment]
Yeah, I was considering doing a sweep of a bunch of easy bugs too. It's good to have them, but 100+ open issues isn't too great :)

**Pull request merged**